src/db.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2015 Jed Lejosne <lejosnej@ainfosec.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+/**
+ * @file   db.c
+ * @author Jed Lejosne <lejosnej@ainfosec.com>
+ * @date   Thu Jul 30 13:13:27 2015
+ *
+ * @brief  Database interaction
+ *
+ * Functions to read/write the policy from/to the database.
+ * This file should stay as separate as possible from the main
+ * project, to be usable in other programs (like a rule manager)
+ */
+
+#include "db.h"
+
+#define db_log(I, ...) { fprintf(stderr, ##__VA_ARGS__); fprintf(stderr, "\n"); }
+
+xcdbus_conn_t *db_xcbus = NULL; /**< A dbus (libxcdbus) handle, initialized by db_init() */
+
+static char*
+parse_value(char *node_path, char *key)
+{
+  char *value;
+  char path[128];
+
+  sprintf(path, "%s/%s", node_path, key);
```

Comment:
snprintf would be strongly preferred here (this comment applies to remaining uses of sprintf as well)